### PR TITLE
Add fast-xml-parser w/ npm auto-update

### LINKED ISSUE #10463: [Request] Add Fast XML Parser
**Library name:** Fast XML Parser
**Git repository url:** https://github.com/NaturalIntelligence/fast-xml-parser
**npm package url(optional):** https://www.npmjs.com/package/fast-xml-parser
**License(s):** MIT
**Official homepage:**
**Wanna say something? Leave message here:**

=====================
Notes from cdnjs maintainer:
Please read the README.md and CONTRIBUTING.md document first.

You are welcome to add a library via sending pull request,
it'll be faster than just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by clairetsai818**
close #10463, cc @amitguptagwl

@cdnjs/intern3 Please help me review this PR, thanks!
Pull request for issue: #10463 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required):https://github.com/NaturalIntelligence/fast-xml-parser
 * Official website (optional, not the repository):
 * NPM package url (optional):https://www.npmjs.com/package/fast-xml-parser
 * License and its reference:[MIT](https://github.com/NaturalIntelligence/fast-xml-parser/blob/master/LICENSE)
 * GitHub / Bitbucket popularity (required):
   - Count of watchers:2
   - Count of stars:3
   - Count of forks:0
 * NPM download stats (optional):
   - Downloads in the last day:20
   - Downloads in the last week:104
   - Downloads in the last month:478

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by clairetsai818** [comment]
@sufuf3 I revised the commit msg, please help me review this, thanks!

File: ajax/libs/fast-xml-parser/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "command"
+  ],
+  "author": "Amit Gupta (https://github.com/amitguptagwl)",
+  "license": "MIT",
+  "npmName": "fast-xml-parser",
+  "npmFileMap": [
+    {
+      "basePath": "lib",
+      "files": [
+        "parser.*"
```
Comment (extend1994) [inline ajax/libs/fast-xml-parser/package.json]: I think `parser*.(css|map|js)` is a better one.

**Comment by clairetsai818** [comment]
@extend1994 I use your script to test auto-update but succeed.

File: ajax/libs/fast-xml-parser/package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    "command"
+  ],
+  "author": "Amit Gupta (https://github.com/amitguptagwl)",
+  "license": "MIT",
+  "npmName": "fast-xml-parser",
+  "npmFileMap": [
+    {
+      "basePath": "lib",
+      "files": [
+        "parser*.(js|css|map)"
```
Comment (hare1039) [inline ajax/libs/fast-xml-parser/package.json]: I can't fetch files using this rule. Please double confirm.

Comment (clairetsai818) [inline ajax/libs/fast-xml-parser/package.json]: @hare1039 Me either. I found all the directory are empty, but I don't know how to fix it. I've checked that the spelling is correct.

**Comment by clairetsai818** [comment]
@cdnjs/intern3 I revised the filemap, and it's fine to run the auto-update now, please help me review this, thanks!

**Comment by sufuf3** [comment]
@clairetsai818 I think you need to add the other commit and do the following:
1. Add the `index.js` of v1.0.0
2. Rename `index.js` to `parser.js`, and minify it.

Because the v1.0.0 structure is not same with the newest version structure.

**Comment by clairetsai818** [comment]
@sufuf3 I added v1.0.0 manually, please help me review this PR again, thanks!

**Comment by sufuf3** [comment]
I think you need to add `cc #10463` in the second commit message.
You can check some examples which had merged.

**Comment by clairetsai818** [comment]
@sufuf3 I revised the commit msg, please help me review this PR again, thanks!

File: ajax/libs/fast-xml-parser/package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "fast-xml-parser",
+  "version": "2.2.1",
```
Comment (extend1994) [inline ajax/libs/fast-xml-parser/package.json]: The latest version is `2.3.0`

File: ajax/libs/fast-xml-parser/package.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    "command"
+  ],
+  "author": "Amit Gupta (https://github.com/amitguptagwl)",
+  "license": "MIT",
+  "npmName": "fast-xml-parser",
+  "npmFileMap": [
+    {
+      "basePath": "lib",
+      "files": [
+        "parser.+(js|css|map)"
```
Comment (extend1994) [inline ajax/libs/fast-xml-parser/package.json]: Yes, it works now!
I am just wondering if `*parser*.+(js|css|map)` more flexible.

File: ajax/libs/fast-xml-parser/package.json (hunk 3 above)
Comment (clairetsai818) [inline ajax/libs/fast-xml-parser/package.json]: @extend1994 Is it okay to let the bot to update this version for me? I'm not sure.

Comment (extend1994) [inline ajax/libs/fast-xml-parser/package.json]: @clairetsai818 Yes I think so if the auto-update config still works in latest version.

**Comment by clairetsai818** [comment]
@sufuf3 please check if the commit message is okay now, thanks!

**Comment by clairetsai818** [comment]
I revised the commit msg, @cdnjs/intern3 please help me review this, thanks!

**Comment by clairetsai818** [comment]
ping

**Comment by pvnr0082t** [comment]
@clairetsai818 Let's rebase it onto the latest master branch. :+1:

**Pull request merged**